### PR TITLE
Mongo updates

### PR DESCRIPTION
**Pull request by varnerac**
## Proposed Changes

This PR does a few things.

* Bumps the MongoDB driver to the latest from upstream, including [my work to add DNS SRV record support](https://github.com/comtihon/mongodb-erlang/pull/237)
* Adds support in `vmq_diversity` for SRV seed lookup in MongoDB
* Modifies existing MongoDB CT test to authenticate, maybe preventing issues like https://github.com/vernemq/vernemq/issues/1678 from popping up in the future
* This probably fixes https://github.com/vernemq/vernemq/issues/1678
* Adds a Docker Compose file to make local integration testing easier
* Adds a new test profile for CI because not everything passes in GitHub Actions right now

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes issue #1678)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, styles...)
- [x] DevOps (Build scripts, pipelines...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [x] I have read the `CODE_OF_CONDUCT.md` document
- [x] All tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if needed)
- [x] Any dependent changes have been merged and published in related repositories
- [x] I have updated changelog (At the bottom of the release version)
- [x] I have squashed all my commits into one before merging

## Further Comments

This bumps `rebar3` like https://github.com/vernemq/vernemq/pull/1731 did. I am happy to back that part out if needed.

The CI profile is a temporary workaround to get us some more testing in CI. Eventually we need all the things tested in Actions. Then, we can ditch the CI profile in for the all tests profile. However, I think this is definitely better than the status quo.

I'll add the DNS SRV documentation if this gets approved. The DNS SRV stuff and the latest MongoDB driver will help if we tackle https://github.com/vernemq/vernemq/issues/1471.

This also probably fixes https://github.com/vernemq/vernemq/issues/1117. I tested my MongoDB changes DNS SRV explicitly against Atlas Cloud databases and it worked in driver tests.

There's no good way to test the SRV functionality in CI because it depends on DNS records.

The diversity suite completely passes locally with the Docker Compose file.

```
varnerac@Drews-MacBook-Pro vernemq % ./rebar3 as all_tests ct --suite apps/vmq_diversity/test/vmq_diversity_provider_SUITE.erl
===> Verifying dependencies...
===> Running neotoma...
===> Analyzing applications...
===> Compiling vmq_acl
===> Compiling vmq_diversity
===> Compiling vmq_generic_msg_store
===> Compiling vmq_mqtt5_demo_plugin
===> Compiling vmq_passwd
===> Compiling vmq_plugin
===> Compiling vmq_plumtree
===> Compiling vmq_pulse
===> Compiling vmq_swc
===> Compiling vmq_webhooks
===> Compiling vmq_ql
===> Compiling vmq_server
===> Compiling vmq_commons
===> Compiling vmq_bridge
cc   -L/usr/local/opt/openssl/lib -I/usr/local/opt/openssl/include vmq_passwd.c -lcrypto -o ../priv/vmq_passwd
===> Running Common Test suites...
%%% vmq_diversity_provider_SUITE: ............
All 12 tests passed.
```

### CODE REVIEW

**Comment by ioolkos** [comment]
very welcome improvements on MongoDB :) 
review upcoming tomorrow, thanks!

**Comment by varnerac** [comment]
That's as much as I can fix in the the CI runs now. If you re-run, it should pass

File: apps/vmq_diversity/src/vmq_diversity_sup.erl [hunk 1]
```diff
@@ -108,12 +108,14 @@ start_all_pools([{mongodb, ProviderConfig}|Rest], Acc) ->
...
+                                  lists:keydelete(max_overflow, 1, PoolOpts)),
     StartFun = fun() ->
-                       Ssl = proplists:get_value(ssl, WorkerArgs, false),
-                       SslOpts = proplists:get_value(ssl_opts, WorkerArgs),
-                       mc_worker:start_link([{ssl,Ssl}, {ssl_opts, SslOpts}|WorkerArgs])
+                       Ssl = proplists:get_value(ssl, WorkerArgs0, false),
+                       SslOpts = proplists:get_value(ssl_opts, WorkerArgs0),
+                       WorkerArgs1 = proplists:delete(ssl, WorkerArgs0),
+                       WorkerArgs2 = proplists:delete(ssl_opts, WorkerArgs1),
+                       mc_worker_api:connect([{ssl,Ssl}, {ssl_opts, SslOpts}|WorkerArgs2])
```
Comment (ioolkos) [inline apps/vmq_diversity/src/vmq_diversity_sup.erl:118]: does `mc_worker_api:connect/1` change anything in behaviour? vs. `mc_worker:start_link/1`? 
(fine with me, I'm just wondering what the difference is as the params are the same.

**Review by ioolkos — APPROVED** [review]
Jep, good for me including the rebar3 bump (worst case I can always go change that).
`pkg_hash_ext` is new in `rebar.lock`
Have not functionally tested the MongoDB stuff but looks good to merge.

File: apps/vmq_diversity/src/vmq_diversity_sup.erl (hunk 1 above)
Comment (varnerac) [reply apps/vmq_diversity/src/vmq_diversity_sup.erl:118]: Yeah. It's a [thin wrapper](https://github.com/comtihon/mongodb-erlang/blob/master/src/api/mc_worker_api.erl#L59-L71) around `mc:worker:start_link/1`.

If you pass `login` and `password` in the proplist, it authenticates the connection.

https://github.com/comtihon/mongodb-erlang/blob/master/src/api/mc_worker_api.erl#L64-L68

I am not sure how authentication was working without this.

**Pull request merged**